### PR TITLE
Migrate to GitHub actions

### LINKED ISSUE #37: Migrate from travis-ci to GitHub actions

### PR DESCRIPTION
**Pull request by kohrVid**
Closes #37.

Similar to changes made to the [core repo](https://github.com/rayyansys/rayyan-formats-core/pull/16), this PR introduces
workflows for each of the supported ruby versions and a composite action that will run the test suite. SimpleCov/Coveralls configuration has also been updated to accommodate this change.

Workflows have been set up to run rspec tests with the following rubies:

* 2.3 - passing
* 2.4 - passing
* 2.5 - passing
* 2.6 - passing
* 2.7 - passing

An example of the CI job can be found on my fork of this repo [here](https://github.com/kohrVid/rayyan-formats-plugins/actions/runs/2679217784).

### CODE REVIEW

**Comment by hammady** [comment]
@kohrVid Please note the edit in the PR description. Writing "Closes #number" automatically closes the issue once the PR is merged.

File: .github/workflows/ruby_2.3.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: 'Ruby 2.3 test'
```
Comment (hammady) [inline .github/workflows/ruby_2.3.yml:1]: How is this strategy different from all-in-one-file like the other repo?

File: .coveralls.yml [hunk 2]
```diff
@@ -1 +1 @@
-service_name: travis-ci
+service_name: github
```
[the remark this answers is not in the document]
Comment (hammady) [reply .coveralls.yml:1]: In another repo, it is called `github-actions`: https://github.com/vagrant-libvirt/vagrant-libvirt/commit/66d394d42e014a1b36f33b603316843f6da0d163.

File: .github/workflows/ruby_2.3.yml (hunk 1 above)
Comment (kohrVid) [reply .github/workflows/ruby_2.3.yml]: I decided to combine the separate workflows after I'd raised this PR yesterday and so hadn't gotten around to updating this until now. I think it makes sense to only use one workflow going forwards

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: 'Test'
+on: [push]
```
Comment (hammady) [inline .github/workflows/tests.yml:2]: Copy settings from the other PR.

File: .coveralls.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hammady) [reply .coveralls.yml]: What is the correct value here?

[the remark this answers is not in the document]
Comment (hammady) [reply .coveralls.yml]: @kohrVid

[the remark this answers is not in the document]
Comment (kohrVid) [reply .coveralls.yml]: The [documentation for the Coveralls Github Action](https://github.com/marketplace/actions/coveralls-github-action) doesn't mention that file so I'm not sure if it's actually in use anymore. When I checked the source code for the [ruby gem](https://github.com/lemurheavy/coveralls-ruby/blob/master/lib/coveralls/configuration.rb), I couldn't really find any mention of a Github Actions `service_name` value

[the remark this answers is not in the document]
Comment (hammady) [reply .coveralls.yml]: Let's remove it then.

[the remark this answers is not in the document]
Comment (kohrVid) [reply .coveralls.yml]: Cool. I've just removed that file and the coveralls gem so hopefully this PR should be ready to merge now

File: Gemfile.lock [hunk 4]
```diff
@@ -13,18 +13,12 @@ GEM
...
     coderay (1.1.2)
-    coveralls (0.8.23)
-      json (>= 1.8, < 3)
-      simplecov (~> 0.16.1)
-      term-ansicolor (~> 1.3)
-      thor (>= 0.19.4, < 2.0)
-      tins (~> 1.6)
     diff-lcs (1.3)
-    docile (1.3.2)
+    docile (1.1.5)
```
Comment (hammady) [inline Gemfile.lock:17]: Downgraded.

File: Gemfile.lock [hunk 5]
```diff
@@ -48,18 +42,12 @@ GEM
       rspec-support (~> 3.9.0)
     rspec-support (3.9.2)
     rubyzip (1.1.7)
-    simplecov (0.16.1)
-      docile (~> 1.1)
+    simplecov (0.14.1)
```
Comment (hammady) [inline Gemfile.lock:45]: Downgraded.

**Review by hammady — APPROVED** [review]

**Pull request merged**